### DISCUSSION TITLE
Post installation configuration

### DISCUSSION THREAD

**Discussion by SteveBz** [1coderookie/KobraGoNeoInsights#7] [General]
Hi Ulf,

So eventually I did have to read the 'tons' of stuff.  And I did open a Reddit account, even :)  The Config stuff was a bit of a nightmare.  

Here:
[https://www.klipper3d.org/Config_checks.html](url)

This was very comprehensive, but not always in the right order.  The Bed mesh calibration was hard and I simplified it with a 3 x 3 bed with only 1 sample.  That speeded it up.  When I have it working I'll do the full one.

The z-offset was also very hard and needs you to get your head round what it actually means.  Eg what is z=0, where is the z-endswitch etc etc.  It took me about 2 days.

Eventually I got though all this.  One of the tricks was to make sure calibrated values were reset to default values.  Calibration to your machine will reset them anyway.

Finally, now I can print.  I get a stunning first layer, really beautiful;, but then the z-axis doesn't move and the base layer just builds up and up making it lumpy.  However, when not printing z works perfectly.  I've calibrated it and it's 100.0 mm perfect. But nothing when printing.  Any ideas?

Here's the standard STL square tower from the klipper website:

![WhatsApp Image 2023-03-16 at 14 19 48](https://user-images.githubusercontent.com/1646078/225647067-f997ad73-b9dc-4bb6-9b85-f79a44b549cf.jpg)

As you can see it just spreads over the base layer.

Kind regards

Steve.

**Comment by 1coderookie** [comment]
Hi Steve,
dunno what's the problem here - maybe something wrong in the slicer settings? No idea, maybe better open up a post on reddit in r/klippers
Setting the z-offset is pretty much genius in Klipper imho, as you don't need to mess around with that in the config, you just do the procedures described in the docs. Once for the probe z-offset -> save it. Once for the endstop z-offset -> save it. So did you do it this way or did you try to evaluate the value and set it as a fix value in your config? I guess you tried it doing it the second way, that's probably why it took you so long.

**Comment by SteveBz** [comment]
I did it both ways. The proper way using probe_calibrate and test was great. I agree genius. But it doesn't work if you don't have the position_min and the position_endstop set properly. I did it my my results weren't saved because they were less than one of those, the end stop I think. I don't remember which one. 

⁣Get BlueMail for Android ​

**Reply by 1coderookie** [reply]
Z position_endstop is also auto generated, I have it deactivated in the manual section.

**Comment by SteveBz** [comment]
Whose printer.cfg file did you use? I used codeneobee's but maybe one of the others  was better. 

⁣Get BlueMail for Android ​

**Reply by 1coderookie** [reply]
The one from @cringegnere and then I adjusted it to my needs. Dunno about the other ones, didn't really looked at them and/or compared them..

**Comment by SteveBz** [comment]
I might try regenerating it. 

Tx. 

⁣Get BlueMail for Android ​

**Comment by SteveBz** [comment]
Hi Ulf,

How are you? It's a sunny morning here today.

I've given up for the time being.  I've had to reset the Neo to Marlin.

I've got 2 sets of SD cards, so I can switch back and forth easily.  The last time, at least I had somehow got my z-offset working, now I can't even do that.  I use the wonderful PROBE_CALIBRATE/TESTZ and it says (for instance), -1.79 so I say ACCEPT and then it says z-offet = 1.699 saved.  Why did it not save -1.79?  So I think, well maybe that's right, so I try to print and it's about 3mm away (ie 1.699 - (-1.79)).  I don't understand why it finds one offset and saves another.

I probably need to go back to Reddit.

Have a good weekend.

Kind regards

Steve.

**Reply by 1coderookie** [reply]
Hi Steve, 
sorry to hear that you switched back to stock-fw. I really don't know what's going on there with your settings and why it doesn't workout, dunno if or what you're maybe doing wrong :( 
I just had a quick look at the printer.cfg from @codeneobee you said you were using and imho the problem with that one is, that the whole endsection where the auto-generated settings are saved, isn't there. Dunno if Klipper re-generates them once you do the procedures tho, you might wanna check that with your own cfg file.  
I only can say that here at my setup it works absolutely reliable. Setting manual z-offset for both the probe and the endstop works flawless. Maybe there is a bigger z-offset somewhere in your config, where it applies your new negative z-offset to, but at the end it's still not sufficient? Dunno..
The only thing I can suggest is to open both your config and the Klipper doc and then read the descriptions of *all* settings to understand what they do and what they are there for. Same for the general procedures like z-offset and so on. And yeah, then maybe open up a reddit post, as I think there are more people with more experience of Klipper than I am. I just read the docs and the descriptions about the settings as mentioned and learned by trial&error, so I can't really help you with specific problems, sorry..

**Comment by SteveBz** [comment]
Hi Ulf,
Thanks for your kind email. I pretty much agree with you. I'm quite busy at work, so at least the Neo's in an operational mode - even if not Klipper.
Next time I have a free afternoon I'll have another go.
I also have an old kit-built 3D called the iMakr Startt, which I have to manage every minor thing. I've made a lot of mods to that, even changing some of the electronic components, like the Heatbed and the associated MOSFETs. The Repetier firmware is much harder than Klipper and you have to inject it right into the board - that's why I was asking at the beginning about burning the bootloader.
Anyhow, they are both working nicely, just not with Klipper.
I've bought a filament runout sensor for less than 5 Euros, OctoPi has pluggins for that, so I thought I'd try that too. Here:
https://www.ebay.co.uk/itm/184993415037
Have a nice day.
Kind regards,
Steve.

**Reply by 1coderookie** [reply]
Sounds good!
Would've been better tho if you'd have this one posted in the belongiong discussion, maybe you can move it?

**Reply by SteveBz** [reply]
Done.

**Reply by 1coderookie** [reply]
Great, thanks!
